test/tls_test.go

```diff
@@ -0,0 +1,175 @@
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend": "LOCAL",
+			"pachd.tls.enabled":     "true",
+			"pachd.tls.secretName":  "blah",
+		},
+	}
+
+	manifest := helm.RenderTemplate(t, options, helmChartPath, "secret", []string{"templates/pachd/deployment.yaml"})
+	fmt.Println(manifest)
+}
+
+func TestEnableDashTLSNoName(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend": "LOCAL",
+			"dash.tls.enabled":      "true",
+			"dash.url":              "http://blah.com",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "secret", []string{"templates/dash/ingress.yaml"})
+	if err == nil {
+		t.Error("Template should error")
+	}
+}
+
+func TestEnableDashTLSExistingSecret(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend": "LOCAL",
+			"dash.tls.enabled":      "true",
+			"dash.tls.secretName":   "blah",
+			"dash.url":              "http://blah.com",
+		},
+	}
+
+	manifest := helm.RenderTemplate(t, options, helmChartPath, "secret", []string{"templates/dash/ingress.yaml"})
+
+	fmt.Println(manifest)
+}
+
+func TestDashTLSNewSecretNoEnable(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend":  "LOCAL",
+			"dash.tls.newSecret.crt": "blah",
+			"dash.url":               "http://blah.com",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "secret", []string{"templates/dash/ingress.yaml"})
+	if err == nil {
+		t.Error("Template should error")
+	}
+}
+
+func TestPachTLSNewSecretNoEnable(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend":   "LOCAL",
+			"pachd.tls.newSecret.crt": "blah",
+		},
+	}
+
+	_, err := helm.RenderTemplateE(t, options, helmChartPath, "deployment", []string{"templates/pachd/deployment.yaml"})
+	fmt.Println(err)
+	if err == nil {
+		t.Error("Template should error")
+	}
+}
+
+func TestPachTLSNewSecret(t *testing.T) {
+	helmChartPath := "../pachyderm"
+	secretName := "newsecret1234"
+	key := "blah"
+	crt := "blah"
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"pachd.storage.backend":      "LOCAL",
+			"pachd.tls.enabled":          "true",
+			"pachd.tls.newSecret.create": "true",
+			"pachd.tls.newSecret.crt":    crt,
+			"pachd.tls.newSecret.key":    key,
+			"pachd.tls.secretName":       secretName,
+		},
+	}
+	manifests := []string{
+		"templates/pachd/tls-secret.yaml",
+		"templates/pachd/deployment.yaml",
+	}
+
+	renderedManifest := helm.RenderTemplate(t, options, helmChartPath, "deployment", manifests)
+	objects, _ := manifestToObjects(renderedManifest)
+
+	for _, obj := range objects {
+		switch obj := obj.(type) {
+		case *v1.Secret:
```

Comment:
This is the kind of validation I'd recommend for the tests above that just print the manifest.  I think it would make sense to refactor your `Test*ExistingSecret` and this test into a table-driven test, and make each of those a case.